OpenRA.Mods.Common/Traits/CustomOwnerLostAction.cs

```diff
@@ -0,0 +1,56 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public enum OwnerLostAction { ChangeOwner, Dispose, Nothing }
+
+	[Desc("This actor does something other than dying when its owner loses.")]
+	public class CustomOwnerLostActionInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[Desc("What does this unit do when its owner loses.",
+		"Allowed values are 'ChangeOwner', 'Dispose', 'Nothing'")]
+		public readonly OwnerLostAction Action = OwnerLostAction.Nothing;
+
+		[Desc("Map player to use when 'Action' is 'ChangeOwner'.")]
+		public readonly string Owner = "Neutral";
+
+		[Desc("Changes the effective (displayed) owner of the actor to the old owner.")]
+		public readonly bool EffectiveOwnerFromOwner = false;
+
+		public object Create(ActorInitializer init) { return new CustomOwnerLostAction(init, this); }
+	}
+
+	public class CustomOwnerLostAction : IEffectiveOwner, INotifyOwnerChanged
```

Comment:
This needs to implement `IDeathActorInitModifier` and add an `EffectiveOwnerInit` to the `TypeDictionary` if `EffectiveOwnerFromOwner` is true.  This should hopefully fix your "Help needed"!